Redundant content-type metadata

There is a group of tests that is failed due to redundant `Content-Type` field in the responses, e.g.

```
    eq(key1_metadata, response['Metadata'])
AssertionError: {'foo': 'bar'} != {'Content-Type': 'text/bla', 'foo': 'bar'}
```
or
```
    eq(metadata, response['Metadata'])
AssertionError: {'key1': 'value1', 'key2': 'value2'} != {'Content-Type': 'audio/ogg', 'key1': 'value1', 'key2': 'value2'}
```

List of affected tests:
```
s3tests_boto3.functional.test_s3:test_object_copy_to_itself_with_metadata
s3tests_boto3.functional.test_s3.test_object_copy_retaining_metadata
s3tests_boto3.functional.test_s3.test_object_copy_replacing_metadata
s3tests_boto3.functional.test_s3.test_object_copy_versioning_multipart_upload
s3tests_boto3.functional.test_s3.test_multipart_upload_resend_part
```

--- a/api/headers.go
+++ b/api/headers.go
@@ -2,15 +2,16 @@ package api
 
 // Standard S3 HTTP request/response constants.
 const (
-	MetadataPrefix       = "X-Amz-Meta-"
-	AmzMetadataDirective = "X-Amz-Metadata-Directive"
-	AmzVersionID         = "X-Amz-Version-Id"
-	AmzTaggingCount      = "X-Amz-Tagging-Count"
-	AmzTagging           = "X-Amz-Tagging"
-	AmzDeleteMarker      = "X-Amz-Delete-Marker"
-	AmzCopySource        = "X-Amz-Copy-Source"
-	AmzCopySourceRange   = "X-Amz-Copy-Source-Range"
-	AmzDate              = "X-Amz-Date"
+	MetadataPrefix            = "X-Amz-Meta-"
+	NeoFSSystemMetadataPrefix = "S3-"
+	AmzMetadataDirective      = "X-Amz-Metadata-Directive"
+	AmzVersionID              = "X-Amz-Version-Id"
+	AmzTaggingCount           = "X-Amz-Tagging-Count"
+	AmzTagging                = "X-Amz-Tagging"
+	AmzDeleteMarker           = "X-Amz-Delete-Marker"
+	AmzCopySource             = "X-Amz-Copy-Source"
+	AmzCopySourceRange        = "X-Amz-Copy-Source-Range"
+	AmzDate                   = "X-Amz-Date"
 
 	LastModified       = "Last-Modified"
 	Date               = "Date"
@@ -87,4 +88,14 @@ var ResponseModifiers = map[string]string{
 	"response-cache-control":       CacheControl,
 	"response-content-disposition": ContentDisposition,
 	"response-content-encoding":    ContentEncoding,
+}
+
+var SystemMetadata = map[string]struct{}{
+	Date:               {},
+	CacheControl:       {},
+	ContentDisposition: {},
+	ContentLength:      {},
+	ContentType:        {},
+	LastModified:       {},
+	ETag:               {},
 }

--- a/api/layer/layer.go
+++ b/api/layer/layer.go
@@ -165,17 +165,6 @@ type (
 		SessionToken *session.Container
 	}
 
-	// PutSystemObjectParams stores putSystemObject parameters.
-	PutSystemObjectParams struct {
-		BktInfo  *data.BucketInfo
-		ObjName  string
-		Metadata map[string]string
-		Prefix   string
-		Reader   io.Reader
-		Size     int64
-		Lock     *data.ObjectLock
-	}
-
 	// ListObjectVersionsParams stores list objects versions parameters.
 	ListObjectVersionsParams struct {
 		BktInfo         *data.BucketInfo

--- a/api/layer/object.go
+++ b/api/layer/object.go
@@ -613,7 +613,8 @@ func (n *layer) getAllObjectsVersions(ctx context.Context, bkt *data.BucketInfo,
 }
 
 func IsSystemHeader(key string) bool {
-	return strings.HasPrefix(key, "S3-")
+	_, ok := api.SystemMetadata[key]
+	return ok || strings.HasPrefix(key, api.NeoFSSystemMetadataPrefix)
 }
 
 func shouldSkip(node *data.NodeVersion, p allObjectParams, existed map[string]struct{}) bool {

